agent/hcp/telemetry/otel_sink.go

```diff
@@ -228,17 +223,30 @@ func (o *OTELSink) flattenKey(parts []string) string {
 	return buf.String()
 }
 
-// toAttributes converts go metrics Labels into OTEL format []attributes.KeyValue
-func toAttributes(labels []gometrics.Label) []attribute.KeyValue {
-	if len(labels) == 0 {
-		return nil
+// labelsToAttributes converts go metrics and provider labels into OTEL format []attributes.KeyValue
+func (o *OTELSink) labelsToAttributes(goMetricsLabels []gometrics.Label) []attribute.KeyValue {
```

Comment:
We need to handle adding the provider labels at the metric level, in addition to the go metrics label. This function merges both into OTEL attribute key value pairs.